### PR TITLE
Fix localTrainingService stream

### PR DESCRIPTION
**Pull request by SparkSnail**
fix issue https://github.com/Microsoft/nni/issues/867

### CODE REVIEW

File: src/nni_manager/training_service/local/localTrainingService.ts [hunk 1]
```diff
@@ -345,6 +345,17 @@ class LocalTrainingService implements TrainingService {
...
             this.onTrialJobStatusChanged(trialJob, oldStatus);
+            //if job is not running, destory job stream
+            if(['SUCCEEDED', 'FAILED', 'USER_CANCELED', 'SYS_CANCELED', 'EARLY_STOPPED'].includes(newStatus)) {
+                if(this.jobStreamMap.has(trialJob.id)) {
+                    const stream = this.jobStreamMap.get(trialJob.id);
+                    if(!stream) {
+                        throw new Error(`Could not find stream in trial ${trialJob.id}`);
+                    }
+                    stream.destroy();
+                    this.jobStreamMap.delete(trialJob.id);
```
Comment (chicm-ms) [inline src/nni_manager/training_service/local/localTrainingService.ts]: This is not a good place to destroy job stream ,  it is better to put it into onTrialJobStatusChanged.

Comment (SparkSnail) [reply src/nni_manager/training_service/local/localTrainingService.ts]: moved to onTrialJobStatusChanged.

File: src/nni_manager/training_service/local/localTrainingServiceForGPU.ts [hunk 2]
```diff
@@ -78,6 +78,17 @@ class LocalTrainingServiceForGPU extends LocalTrainingService {
...
     protected onTrialJobStatusChanged(trialJob: LocalTrialJobDetailForGPU, oldStatus: TrialJobStatus): void {
+        //if job is not running, destory job stream
+        if(['SUCCEEDED', 'FAILED', 'USER_CANCELED', 'SYS_CANCELED', 'EARLY_STOPPED'].includes(trialJob.status)) {
+            if(this.jobStreamMap.has(trialJob.id)) {
+                const stream = this.jobStreamMap.get(trialJob.id);
+                if(!stream) {
+                    throw new Error(`Could not find stream in trial ${trialJob.id}`);
+                }
+                stream.destroy();
+                this.jobStreamMap.delete(trialJob.id);
```
Comment (chicm-ms) [inline src/nni_manager/training_service/local/localTrainingServiceForGPU.ts]: I assume this is not specific for GPU,   looks like should move to parent class.

Comment (SparkSnail) [reply src/nni_manager/training_service/local/localTrainingServiceForGPU.ts]: moved to parent class

**Pull request merged**